### PR TITLE
Add System Tests for Opening Python and Cpp Interfaces

### LINKED ISSUE #28543: Add Automated Test for opening Interfaces
In pre-release Smoke testing, we often find an interface doesn't open. 

Having a test which checks that each interface opens (and closes) correctly would be therefore be very useful!

### PR DESCRIPTION
**Pull request by robertapplin**
~Merge after #29627~
---------------------

**Description of work.**
This PR introduces two system tests that check that the python and cpp interfaces open ok.

Note that I needed to move the `QECoverageGUI` class to a different folder in order to be able to run the startup script for `QECoverage` successfully without an error.

**To test:**
Make sure that the two tests `PythonInterfacesStartupTest` and `CppInterfacesStartupTest` look reasonable and pass.

Fixes #28543

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DannyHindson — COMMENTED** [review]
Just started having a look at this and for some reason the PythonInterfacesStartUp system test is failing when run on my local Windows machine. Not sure if it's failing now because something has gone into master relating to the engineering UI since your PR passed it checks or whether it's something Windows-related. Any idea?

image

**Review by DannyHindson — COMMENTED** [review]
Actually - forget that. Just realised the problem was happening because I hadn't rebuilt the Qt5 target. It works now. Apologies if you'd spent any time looking into this

**Comment by robertapplin** [comment]
> Actually - forget that. Just realised the problem was happening because I hadn't rebuilt the Qt5 target. It works now. Apologies if you'd spent any time looking into this

Thats good, no worries I was away from my desk for lunch anyways!

**Review by DannyHindson — COMMENTED** [review]
This all looks fine to me. I'll approve once the earlier PR #29627 is approved and the diff reduces down to the changes made in the PR only

**Comment by robertapplin** [comment]
> This all looks fine to me. I'll approve once the earlier PR #29627 is approved and the diff reduces down to the changes made in the PR only

#29627 has now been merged so the diff should be only for this PR now

**Review by DannyHindson — CHANGES_REQUESTED** [review]
Just one small question about the feature usage string not having the truncation specifiers in it any more

File: scripts/Interface/reduction_gui/reduction/scripter.py
Comment (robertapplin) [reply scripts/Interface/reduction_gui/reduction/scripter.py:416]: I think you are right, this should have only been a temporary change when I was debugging 👍

**Review by DannyHindson — APPROVED** [review]
Looks good thanks

File: Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py
Comment (martyngigg) [inline Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py:18]: Just a quick question about `Frequency_Domain_Analysis`. I remember we discussed it but was it not the Qt4/5 issue or was it something else?

Comment (robertapplin) [reply Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py:18]: For this interface it appears to be something else. I have tried to debug it but I couldn't figure out what was going on.

I think someone with more experience or better knowledge of the Muon area might be able to work it out. Can I create a separate maintenance issue for this to be fixed and incorporated into this test?

Comment (martyngigg) [reply Testing/SystemTests/tests/qt/PythonInterfacesStartupTest.py:18]: Yes please do create a separate issue. We need the tests to be reliable indicators of if something is wrong so it would be good to figure this out.

**Comment by robertapplin** [comment]
This is the issue for adding `Frequency_Domain_Analysis.py` to the `PythonInterfacesStartupTest`: #29773

**Pull request merged**
